crates/nu-parser/src/parser.rs

```diff
@@ -954,7 +954,12 @@ pub fn parse_internal_call(
     let _ = working_set.add_span(call.head);
 
     let decl = working_set.get_decl(decl_id);
-    let signature = decl.signature();
+    let signature = if let Some(block_id) = decl.block_id() {
```

Comment:
Would you mind adding `get_signature()` to `StateWorkingSet` instead? Seems like it would be nice to have the same method there.

Reply:
absolutely, I was thinking of doing that but then wasn't sure if it would belong in this PR. will add it now

Reply:
If you find some cases where getting the block's signature from the working set improves something, feel free to refactor it to use `working_set.get_signature(decl)`. Preferably in a separate PR, let's keep this one focused.